### ISSUE TITLE
Interactive rebase

### ISSUE THREAD

**Issue by pascalberger** [gmasterscm/ama#18]
Any plans for adding a UI for interactive rebases (for squashing or reordering commits or changing commit messages)?

**Comment by psantosl** [comment]
Yes!
I've been working on this today, indeed :-P, studying how we are going to do it and coming up with some proposals.

Thoughts? :P

![image](https://user-images.githubusercontent.com/380766/30484226-0b4d8bbe-9a2a-11e7-9277-4805e4e8a03f.png)

By the way, what is your favorite UI for dealing with interactive rebases now? SourceTree? Just want to double check.

**Comment by pascalberger** [comment]
@psantosl Would this also work for squashing, reordering and editing commit messages?

Currently I'm mostly using SourceTree, which IMHO has a quite nice and intuitive UI for interactive rebases (beside its overall slowness and bugs since 2.x)

**Comment by psantosl** [comment]
Yes, that's the idea: it would be straightforward to mark the action of each commit: pick, squash, reword, whatever... Reorder will be drag and drop.

SourceTree, ok, got it. I found it nearly unusable :P but it must be by its lack of visualization :-)

**Comment by AngusMcIntyre** [comment]
I find Sourcetree's interactive rebase UI horrible. Its hard to know exactly how it will behave and tricky to undo without totally restarting.

It would be fun to be able to drag and drop commit to re-order and squash. When squashing, I'd also like an orderable list on the 'squash' commit.
